src/lib/repl_service_ctx.cpp

```diff
@@ -64,6 +64,18 @@ repl_service_ctx::repl_service_ctx(nuraft::raft_server* server) : _server(server
 
 bool repl_service_ctx::is_raft_leader() const { return _server->is_leader(); }
 
+const std::string& repl_service_ctx::raft_leader_id() const {
+    // when adding member to raft,  the id recorded in raft is a hash
+    // of passed-in id (new_id in add_member()), the new_id was stored
+    // in endpoint field.
+    static std::string const empty;
+    if (!_server) return empty;
```

Comment:
do we need to define a static variable `empty`? Just return `""` or `std::string()`